internal/resources/smbshare.go

```diff
@@ -148,146 +148,31 @@ func (m *SmbShareManager) Update(
-		if planner.IsClustered() && b != clusteredBackend {
-			err = fmt.Errorf(
-				"Can not convert SmbShare to clustered instance."+
-					" Current backend: %s",
-				b)
-			m.logger.Error(
-				err,
-				"Backend inconsistency detected",
-				"SmbShare.Namespace", instance.Namespace,
-				"SmbShare.Name", instance.Name,
-				"SmbShare.UID", instance.UID)
-			return Result{err: err}
+		if result := m.updateBackend(ctx, planner); result.Yield() {
+			return result
 		}
-		if !planner.IsClustered() && b != standardBackend {
-			err = fmt.Errorf(
-				"Can not convert SmbShare to non-clustered instance."+
-					" Current backend: %s",
-				b)
-			m.logger.Error(
-				err,
-				"Backend inconsistency detected",
-				"SmbShare.Namespace", instance.Namespace,
-				"SmbShare.Name", instance.Name,
-				"SmbShare.UID", instance.UID)
-			return Result{err: err}
+	} else {
+		if result := m.validateBackend(planner); result.Yield() {
+			return result
 		}
 	}
 
 	if planner.IsClustered() {
-		if !planner.MayCluster() {
-			err = fmt.Errorf(
-				"CTDB clustering not enabled in ClusterSupport: %v",
-				planner.GlobalConfig.ClusterSupport)
-			m.logger.Error(err, "Clustering support is not enabled")
-			return Result{err: err}
-		}
-		_, created, err := m.getOrCreateStatePVC(
-			ctx, planner, destNamespace)
-		if err != nil {
-			return Result{err: err}
-		} else if created {
-			m.logger.Info("Created shared state PVC")
-			return Requeue
-		}
-
-		statefulSet, created, err := m.getOrCreateStatefulSet(
-			ctx, planner, destNamespace)
-		if err != nil {
-			return Result{err: err}
-		} else if created {
-			// StatefulSet created successfully - return and requeue
-			m.logger.Info("Created StatefulSet")
-			m.recorder.Eventf(instance,
-				EventNormal,
-				ReasonCreatedStatefulSet,
-				"Created stateful set %s for SmbShare", statefulSet.Name)
-			return Requeue
-		}
-
-		resized, err := m.updateStatefulSetSize(
-			ctx, statefulSet,
-			int32(planner.SmbShare.Spec.Scaling.MinClusterSize))
-		if err != nil {
-			return Result{err: err}
-		} else if resized {
-			m.logger.Info("Resized statefulSet")
-			return Requeue
+		result := m.updateClusteredState(ctx, planner)
+		if result.Yield() {
+			return result
 		}
 	} else {
-		deployment, created, err := m.getOrCreateDeployment(
-			ctx, planner, destNamespace)
-		if err != nil {
-			return Result{err: err}
-		} else if created {
-			// Deployment created successfully - return and requeue
-			m.logger.Info("Created deployment")
-			m.recorder.Eventf(instance,
-				EventNormal,
-				ReasonCreatedDeployment,
-				"Created deployment %s for SmbShare", deployment.Name)
-			return Requeue
-		}
-
-		resized, err := m.updateDeploymentSize(ctx, deployment)
-		if err != nil {
-			return Result{err: err}
-		} else if resized {
-			m.logger.Info("Resized deployment")
-			return Requeue
+		result := m.updateNonClusteredState(ctx, planner)
+		if result.Yield() {
+			return result
```

Comment:
Code-style consistency: why not have `if result := ... ; result.Yield() {` (within same line), as done in other places?